### PR TITLE
Fixed lot of cpu stuff

### PR DESCRIPTION
**Pull request by Princess-of-Sleeping**
From https://github.com/vitasdk/vita-headers/blob/master/vita.header_warn.cmake#L43

This PR will be merged 2023/10/15 or after.

### CODE REVIEW

File: include/psp2kern/kernel/cpu/atomic.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#define _PSP2KERN_KERNEL_CPU_ATOMIC_H_
+
+#include <psp2kern/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+int ksceKernelAtomicAdd32AndGet64InHiLoRange(unsigned long long *val, int add_val, int limit);
```
Comment (d3m3vilurr) [inline include/psp2kern/kernel/cpu/atomic.h:17]: just a question, don't we need to change types to `SceInt32` and `SceUInt64`??

Comment (Princess-of-Sleeping) [reply include/psp2kern/kernel/cpu/atomic.h:17]: The SceCpuForKernel::atomic stuff was added straight from the wiki. I can change it to SceInt32/SceInt64 if you wish.

Comment (d3m3vilurr) [reply include/psp2kern/kernel/cpu/atomic.h:17]: yep. can you change them?
i'll merge this after that jobs

Comment (Princess-of-Sleeping) [reply include/psp2kern/kernel/cpu/atomic.h]: done

**Pull request merged**